pkg/table/tables/index_test.go

```diff
@@ -168,3 +171,44 @@ func buildTableInfo(t *testing.T, sql string) *model.TableInfo {
 	require.NoError(t, err)
 	return tblInfo
 }
+
+func TestGenIndexValueFromIndex(t *testing.T) {
+	tblInfo := buildTableInfo(t, "create table a (a int primary key, b int not null, c text, unique key key_b(b));")
+	tblInfo.State = model.StatePublic
+	tbl, err := tables.TableFromMeta(lkv.NewPanickingAllocators(0), tblInfo)
+	require.NoError(t, err)
+
+	sessionOpts := encode.SessionOptions{
+		SQLMode:   mysql.ModeStrictAllTables,
+		Timestamp: 1234567890,
+	}
+
+	encoder, err := lkv.NewBaseKVEncoder(&encode.EncodingConfig{
+		Table:          tbl,
+		SessionOptions: sessionOpts,
+	})
+	require.NoError(t, err)
+	encoder.SessionCtx.GetSessionVars().RowEncoder.Enable = true
+
+	data1 := []types.Datum{
+		types.NewIntDatum(1),
+		types.NewIntDatum(23),
+		types.NewStringDatum("4.csv"),
+	}
+	tctx := encoder.SessionCtx.GetTableCtx()
+	_, err = encoder.Table.AddRecord(tctx, data1)
+	require.NoError(t, err)
+	kvPairs := encoder.SessionCtx.TakeKvPairs()
+
+	indexKey := kvPairs.Pairs[1].Key
+	indexValue := kvPairs.Pairs[1].Val
+
+	_, idxID, _, err := tablecodec.DecodeIndexKey(indexKey)
+	require.NoError(t, err)
+
+	idxInfo := model.FindIndexInfoByID(tbl.Meta().Indices, idxID)
+
+	valueStr, err := tables.GenIndexValueFromIndex(indexKey, indexValue, tbl.Meta(), idxInfo)
+	require.NoError(t, err)
+	require.Equal(t, []string{"23"}, valueStr)
```

Comment:
what's the output for binary or bits?

Reply:
`tables.GenIndexValueFromIndex`'s content is retrieved from the original function `genKeyExistsErr` in `ddl`
https://github.com/pingcap/tidb/blob/5745d3df9de4afa98909eb972c8456f8135d65f5/pkg/ddl/index.go#L1601-L1629